### PR TITLE
First stab at spec-validator

### PR DESCRIPTION
**Pull request by andersy005**
Towards #2

TODO

- [x] Validate json
- [x] Validate CSV

### CODE REVIEW

**Comment by andersy005** [comment]
Cc @rabernat, @matt-long  for review if you have time

**Comment by rabernat** [comment]
Wow it looks like a lot of effort went into this! I'm not sure I'll be able to provide a comprehensive review. But I like what you're doing.

Do you check the csv file for consistency with the json file?

File: esmcol_validator/validator.py [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+            return True, None
+        except ValidationError as e:
+            logger.warning('ESMCol Validation Error')
+            return False, f'{e.message} of {list(e.path)}'
+
+        except Exception as e:
+            logger.exception('ESMCol error')
+            return False, f'{e}'
+
+    def validate_csv(self, content):
```
Comment (andersy005) [inline esmcol_validator/validator.py:184]: > Do you check the csv file for consistency with the json file?

Yes. This is where the csv validation is happening. I may be missing some edge cases here, but the more general cases are well covered (I think :))

**Comment by andersy005** [comment]
If this looks good, we can merge it for the time being. I'm happy to fix issues or add new features in a separate PR. 

Once this is merged, I will revisit https://github.com/NCAR/intake-esm/pull/135

**Comment by rabernat** [comment]
🚀

**Pull request merged**